service/src/main/java/bio/terra/workspace/service/workspace/GcpCloudContextService.java

```diff
@@ -160,30 +156,27 @@ public CloudContext makeCloudContextFromDb(DbCloudContext dbCloudContext) {
   public Optional<GcpCloudContext> getGcpCloudContext(UUID workspaceUuid) {
     return workspaceDao
         .getCloudContext(workspaceUuid, CloudPlatform.GCP)
-        .flatMap(GcpCloudContext::deserialize);
+        .map(GcpCloudContext::deserialize);
   }
 
   /**
-   * Retrieve the GCP cloud context. If it does not have the policies filled in, retrieve the
-   * policies from Sam, fill them in, and update the cloud context.
-   *
-   * <p>This is used during controlled resource create. Since the caller may not have permission to
-   * read the workspace policies, we use the WSM SA to query Sam.
+   * Retrieve the GCP cloud context. This is used during controlled resource create.
    *
    * @param workspaceUuid workspace identifier of the cloud context
    * @return GCP cloud context with all policies filled in.
    */
-  public GcpCloudContext getRequiredGcpCloudContext(
-      UUID workspaceUuid, AuthenticatedUserRequest userRequest) throws InterruptedException {
-    return getGcpCloudContext(workspaceUuid)
-        .orElseThrow(
-            () -> new CloudContextRequiredException("Operation requires GCP cloud context"));
+  public GcpCloudContext getRequiredGcpCloudContext(UUID workspaceUuid) {
+    GcpCloudContext cloudContext =
+        getGcpCloudContext(workspaceUuid)
+            .orElseThrow(
+                () -> new CloudContextRequiredException("Operation requires GCP cloud context"));
+    return cloudContext;
```

Comment:
I think this step should also validate that the context is in the READY state. As-is, I think this won't throw an exception here, but callers will expect the context fields to be populated per the javadoc.

Reply:
I did that for all context types, assuming that if we were requiring the context, we would also want it ready. Things broke all over the place. Turns out we "get required" in the middle of many operations, and it would find the state it was operating on (updating, deleting). I added PF-2788 to track adding operation checks on resource state. I could add it to this, but it was already a big change set...